### PR TITLE
feat: add auto release

### PR DESCRIPTION
**Pull request by drodarie**
## What has been added

- Add conventional commits for pre-commits and pull request
- Add release pipeline (similar to bsb-core)
- Add CHANGELOG.md file
- Update README.md
- Add dev installation

## What remains to be done

- [X] Setup OIDC on PYPI (requires admin rights)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -13,12 +13,22 @@ dependencies = ["bsb-core~=4.0"]
...
+    "snakeviz~=2.1",
+    "bump-my-version~=0.24"
+]
 
 [tool.flit.module]
 name = "bsb_test"
 
 [tool.black]
-line-length=120
+line-length = 90
```
Comment (Helveg) [inline pyproject.toml:31]: why change this?

**Review by Helveg — APPROVED** [review]
Why'd you change the line length?

File: pyproject.toml (hunk 1 above)
Comment (drodarie) [reply pyproject.toml:31]: I wanted to match `bsb-core` black configuration. I can revert this, if you prefer to have different configs for each bsb repos.

**Comment by drodarie** [comment]
@Helveg let me know when I can merge this (i.e the OIDC on PYPI is implemented)

**Pull request merged**